app/code/community/Ebizmarts/MailChimp/Model/Resource/Ecommercesyncdata/Customers/Collection.php

```diff
@@ -27,14 +27,16 @@ public function _construct()
     /**
      * @param Mage_Customer_Model_Resource_Customer_Collection $preFilteredCustomersCollection
      */
-    public function joinLeftEcommerceSyncData($preFilteredCustomersCollection, $columns = array('m4m.*'))
+    public function joinLeftEcommerceSyncData($preFilteredCustomersCollection)
     {
         $mailchimpTableName = $this->getMailchimpEcommerceDataTableName();
+        $joinCondition      = "m4m.related_id = e.entity_id AND m4m.type = '%s' AND m4m.mailchimp_store_id = '%s'";
         $preFilteredCustomersCollection->getSelect()->joinLeft(
-            array('m4m' => $mailchimpTableName),
-            "m4m.related_id = main_table.entity_id AND m4m.type = '" . Ebizmarts_MailChimp_Model_Config::IS_CUSTOMER
-            . "' AND m4m.mailchimp_store_id = '" . $this->getMailchimpStoreId() . "'",
-            $columns
+            array("m4m" => $mailchimpTableName),
+            sprintf($joinCondition, Ebizmarts_MailChimp_Model_Config::IS_CUSTOMER, $this->getMailchimpStoreId() ),
+            array()
```

Comment:
May be better to simply avoid this parameter in stead of sending an empty array.

Reply:
Solved.